Common/Product/SharedProject/ProjectNode.cs

```diff
@@ -2790,9 +2796,14 @@ internal enum EventTriggering {
             if (!IsProjectOpened)
                 return;
 
-            EnvDTE.Project automationObject = GetAutomationObject() as EnvDTE.Project;
+            var solutionBuild = (IVsSolutionBuildManager)GetService(typeof(SVsSolutionBuildManager));
+            IVsProjectCfg[] cfg = new IVsProjectCfg[1];
+            ErrorHandler.ThrowOnFailure(
+                solutionBuild.FindActiveProjectCfg(IntPtr.Zero, IntPtr.Zero, GetOuterHierarchy(), cfg));
 
-            SetConfiguration(Utilities.GetActiveConfigurationName(automationObject));
+            string name;
+            ErrorHandler.ThrowOnFailure(cfg[0].get_CanonicalName(out name));
+            SetConfiguration(name);
```

Comment:
Ok. The new content should be copy paste but I'll make sure the surrounding stuff matches too.